coresimd/src/x86/i586/bmi.rs

```diff
@@ -127,16 +127,6 @@ pub unsafe fn _blsr_u64(x: u64) -> u64 {
     x & (x.wrapping_sub(1))
 }
 
-/// Counts the number of trailing least significant zero bits.
-///
-/// When the source operand is 0, it returns its size in bits.
-#[inline(always)]
-#[target_feature = "+bmi"]
-#[cfg_attr(test, assert_instr(tzcnt))]
-pub unsafe fn _tzcnt_u16(x: u16) -> u16 {
```

Comment:
GCC and clang define these:

```c++
 #define _tzcnt_u16(a)     (__tzcnt_u16((a)))
```

E.g. see here: https://clang.llvm.org/doxygen/bmiintrin_8h_source.html

Reply:
Perhaps, but this isn't on Intel's intrinsic list, which is what this crate is for.

Reply:
Yep, the Intel intrinsic list only defines `_tzcnt_u{32,64}`. 

MSVC does not define it here: https://msdn.microsoft.com/en-us/library/hh977023.aspx

and the Intel C++ compiler doesn't seem to define it either. The GCC compiler defines it The GCC definition is here: https://github.com/gcc-mirror/gcc/blob/master/gcc/config/i386/bmiintrin.h#L38M

> Perhaps, but this isn't on Intel's intrinsic list, which is what this crate is for

Intel Intrinsics list doesn't define many x86 intrinsics either (e.g. all of the AMD ones).

Anyways, I am fine with removing it, just saying that two widely used C and C++ compilers offer it, which is why it was offered here.